test/phone_form_field_test.dart

```diff
@@ -80,6 +82,38 @@ void main() {
         await tester.pumpWidget(getWidget(showFlagInInput: false));
         expect(find.byType(CircleFlag), findsNothing);
       });
+
+      testWidgets('Should format when shouldFormat is true', (tester) async {
+        PhoneNumber? phoneNumber = PhoneNumber.parse(
+          '',
+          destinationCountry: IsoCode.FR,
+        );
+
+        await tester.pumpWidget(
+            getWidget(initialValue: phoneNumber, shouldFormat: true));
+        await tester.pump(const Duration(seconds: 1));
+        final phoneField = find.byType(PhoneFormField);
+        await tester.enterText(phoneField, '677777777');
+        await tester.pump(const Duration(seconds: 1));
+        expect(find.text('6 77 77 77 77'), findsOneWidget);
+      });
+      testWidgets('Show or hide dial code', (tester) async {
+        PhoneNumber? phoneNumber = PhoneNumber.parse(
+          '',
+          destinationCountry: IsoCode.FR,
+        );
+
+        await tester.pumpWidget(getWidget(
+            initialValue: phoneNumber,
+            showDialCode: false,
+            defaultCountry: IsoCode.FR));
+        await tester.pump(const Duration(seconds: 1));
+        expect(find.text('+ 33'), findsNothing);
+        // final phoneField = find.byType(PhoneFormField);
```

Comment:
This tests `showDialCode: false`, need a test for `showDialCode: true` too, so this does not break in the future.